lib/theme/index.js

```diff
@@ -28,7 +27,7 @@ function Theme(ctx) {
   languages.push('default');
 
   this.i18n = new I18n({
-    languages: _(languages).compact().uniq().value()
+    languages: [...new Set(languages.filter(Boolean))]
```

Comment:
what about `Array.from(new Set(languages.filter(Boolean)))`?

Reply:
`value()` is a lodash method, which is used to get value from lodash wrapped prototype chain.

Reply:
I have make a benchmark at jsPerf: https://jsperf.com/lodash-uniq-vs-javascript-set/1

It seems the diffrence between `Array.from` & spread syntax is negligible, at least in browser.

Reply:
I see.

The benchmark showed the spread syntax is the source of regression. I noticed you reverted the last commit; to confirm the source?

Reply:
> It seems the diffrence between Array.from & spread syntax is negligible, at least in browser.

Similar result in travis too. Can you try revert back to lodash just for this line? to test whether `Set()` is the culprit.

Reply:
Found another alternative is `languages.filter((v, i, a) => a.indexOf(v) === i)` ([source](https://blog.bitsrc.io/you-dont-need-lodash-or-how-i-started-loving-javascript-functions-3f45791fa6cd)). It's <10% slower in Chrome while faster in FF,

Reply:
I have revert this line back to lodash and found out it is not the culprit. So the `view.js` is the smoking gun.

During the generation process, the `view.js` will be called every time when a page is rendered (#2164). We might need tohandle it carefully.

Reply:
But that doesn't explain why lodash is faster in `view.js`.

Reply:
@curbengh I have done through some investigation about performance of `Object.assign`. It seems that `Object.assign` will meet performance issue when facing large object. Even Node.js itself is still using the deprecated `util._extend`, because it is still a lot more faster than `Object.assign`.

Reply:
A bit more interesting... Even if I replace Object.assign with lodash.assign, the impact is still there. So maybe the problem is at `Object.getPrototypeOf`..

Reply:
That seems like the last suspect now. what about `return Object.assign({}, locals, locals.prototype, data, {` ? ([source](https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#_extend))

Reply:
@curbengh In fact I have tried `local.prototype` at very beginning but it won't pass the test.